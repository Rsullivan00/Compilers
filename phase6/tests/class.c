 int g(int x){
printf("Int func g: %d\n", x);
return x;
}

int h(int y){
printf("In func h: %d\n", y);
return y;
}
int main(void){
int n;
int x;
int y;

y = 4;
printf("%d\n", y);

x = 1+y;
printf("%d\n", x);

n = 3;
printf("%d\n", n);

n = g(h(5));

printf("In main: %d\n", n);

if(n > 0){
if(y > 0){
if(x < 0){
printf("All excpet x are greater than 0\n");
}
else{
printf("All greater than 0\n");
}
}
}

x = 5;
y = 5;
while(x != 0){
while(y != 0){
printf("y is: %d\n", y);
y = y - 1;
}
printf("X is: %d\n", x);
x = x - 1;
}
return 0;
}
