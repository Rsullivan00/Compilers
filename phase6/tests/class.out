4
5
3
In func h: 5
Int func g: 5
In main: 5
All greater than 0
y is: 5
y is: 4
y is: 3
y is: 2
y is: 1
X is: 5
X is: 4
X is: 3
X is: 2
X is: 1
